Initial package for TrueType font Hack

timmooney opened the pull request:
This is my first-ever hipster package, so I would appreciate it if both Alexander and Aurelien would review the Makefile and manifest carefully.  I'll happily take any and all feedback on things that should be done differently, any package information that might be missing, etc.  I used Aurelien's liberation-ttf Makefile and manifest as a starting point, and I've been following both [Building with oi-userland on the wiki](https://wiki.openindiana.org/oi/Building+with+oi-userland) and [Quick start for oi-userland](http://docs.openindiana.org/dev/userland/).

[Hack](http://sourcefoundry.org/hack/) is a beautiful monospaced font descended from BitStream Vera and DejaVu, but with many improvements.  It's designed specifically for viewing source code on a modern display.

Commit:
package TrueType font Hack and add to desktop_common

pyhalov commented on components/fonts/hack-ttf/Makefile:
```diff
@@ -0,0 +1,64 @@
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=      $(COMPONENT_NAME)-$(COMPONENT_VERSION).zip
+COMPONENT_ARCHIVE_URL=	\
+	https://github.com/chrissimpkins/Hack/releases/download/v$(COMPONENT_VERSION)/$(COMPONENT_NAME)-v2_020-ttf.zip
```
Better use $(COMPONENT_ARCHIVE) here, modifying COMPONENT_ARCHIVE so that it fits?

pyhalov commented on components/fonts/hack-ttf/Makefile:
```diff
@@ -0,0 +1,64 @@
+# transform the upstream version into a suitable version number that
+# doesn't have a leading 0 in any component.
+#
+IPS_COMPONENT_VERSION= $(shell echo $(COMPONENT_VERSION) | sed -e 's/\.0*\([1-9]\)/.\1/')
```
Please, shorten comment to one line (or remove) and move IPS_COMPONENT_VERSION just below COMPONENT_VERSION (so that one looking at Makefile immediately could see this magic)

timmooney replied:
The comments are nearly identical to the same comments and placement in sysutils/pwgen, but I've followed your advice and moved IPS_COMPONENT_VERSION closer to COMPONENT_VERSION and reduced the comment to just one line.

I'll push this fix and others in an amended commit once I have all your suggested changes complete.

pyhalov commented on components/fonts/hack-ttf/hack.p5m:
```diff
@@ -0,0 +1,34 @@
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+# Aurelien was manually adding this link for the fonts he packaged, so
+# follow suit.
```
Either say in comment why this link really needed or remove this comment.

pyhalov commented:
Component misses sample-manifest (Do 'gmake sample-manifest'. Add manifests/sample-manifest.p5m to the component).

pyhalov commented:
@alarcher :  Do we want to have this font installed by default?

timmooney commented:
I believe I have fixed each of the issues Alexander mentioned in each one of his review comments.  Thanks for the feedback on my first package contribution!

I've done a git rebase and squashed the additional commits, and force pushed this back to my github branch.

Please review and let me know if there are any additional changes you would like to see made.

Thanks,

Tim

alarcher commented on components/fonts/hack-ttf/Makefile:
```diff
@@ -0,0 +1,61 @@
+	https://github.com/chrissimpkins/Hack/releases/download/v$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_ARCHIVE_HASH= \
+  sha256:048566ae79c580f725b68340d9d2a3b0fa125fb08c1d30cf0a7c327d07ab739a
+COMPONENT_LICENSE=	Hack Open Font License Version 2.0
```
The license file seems to imply that some files are also under the Bitstream Vera License.

timmooney replied:
@alarcher, is there anything I need to do related to your observation about the license?  The hack.license file is included verbatim from the hack source distribution.

Are you suggesting that I should update the COMPONENT_LICENSE in the Makefile to mention both the Hack Open Font License 2.0 and the Bitstream Vera License?

alarcher replied:
@timmooney yes exactly, that's what I was suggesting.

timmooney replied:
Thanks @alarcher for the clarification!

Looking at other components that have multiple licenses, there doesn't seem to be a single method of formatting the value of COMPONENT_LICENSE.  Some, like components/libjepg-turbo, use commas between each of the licenses, while some (components/libodfgen, components/libwpd) just use spaces between the license abbreviations.

Since I won't be abbreviating the licenses, my intention is to use a comma between the two licenses, like this:

`COMPONENT_LICENSE=	Hack Open Font License Version 2.0, BitStream Vera Font License`

alarcher commented on components/fonts/hack-ttf/hack.p5m:
```diff
@@ -0,0 +1,32 @@
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+link path=etc/X11/fontpath.d/hack:pri=42 target=../../../usr/share/fonts/TrueType/hack
```
This should be fine.

alarcher reviewed (commented):
Looking good.

alarcher commented:
@pyhalov @timmooney The font looks elegant so could be indeed installed by default.

pyhalov commented:
Then we can also modify components/meta-packages/install-types/includes/desktop_common and bump revision in components/meta-packages/install-types/Makefile

pyhalov commented:
@timmooney - please, address @alarcher's  comment about license and (optionally) add dependency on it to desktop_common.

timmooney commented:
I have updated COMPONENT_LICENSE as requested.
I added system/font/truetype/hack to meta-packages/install-types/includes/desktop_common, and I updated COMPONENT_REVISION from 8 to 9 in meta-packages/install-types/Makefile

I've rebased and squashed the commits into one and force pushed to my hack branch.

Thanks for the feedback.  Please let me know if there are other changes you believe should be made before this is merged.

alarcher commented:
@timmooney Yes comma separated is the standard. It is all good, thank you!
